Gameplay/MainMenuManager.cpp

```diff
@@ -0,0 +1,204 @@
+	, _state(State::MAIN)
+	, _state_changed(false)
+	, _main_background(nullptr)
+	, _button_background(nullptr)
+	, _button_play(nullptr)
+	, _button_quit(nullptr)
+	, _button_settings(nullptr)
+	, _button_credits(nullptr)
+	, _credits(nullptr)
+	, _settings(nullptr)
+	, _button_back(nullptr)
+{
+}
+
+void Hachiko::Scripting::MainMenuManager::OnAwake()
+{
+	bool any_null = false;
+
+	_main_background = game_object->GetFirstChildWithName(
+		"button_whitebackground");
+	any_null |= (_main_background == nullptr);
+	
+	_button_background = game_object->GetFirstChildWithName(
+		"button_background");
+	any_null |= (_button_background == nullptr);
+
+	_button_play = game_object->GetFirstChildWithName(
+		"button_play")->GetComponent<ComponentButton>();
+	any_null |= (_button_play == nullptr);
+
+	_button_quit = game_object->GetFirstChildWithName(
+		"button_quit")->GetComponent<ComponentButton>();
+	any_null |= (_button_quit == nullptr);
+
+	_button_settings = game_object->GetFirstChildWithName(
+		"button_settings")->GetComponent<ComponentButton>();
+	any_null |= (_button_settings == nullptr);
+
+	_button_credits = game_object->GetFirstChildWithName(
+		"button_credits")->GetComponent<ComponentButton>();
+	any_null |= (_button_credits == nullptr);
+
+	_credits = game_object->GetFirstChildWithName(
+		"credits");
+	any_null |= (_credits == nullptr);
+
+	_settings = game_object->GetFirstChildWithName(
+		"settings");
+	any_null |= (_settings == nullptr);
+
+	_button_back = game_object->GetFirstChildWithName(
+		"button_back")->GetComponent<ComponentButton>();
+	any_null |= (_button_back == nullptr);
+
+	if (any_null)
+	{
+		HE_LOG("If you renamed or deleted any game object in this scene," 
+			   " please check this script.");
+	}
+}
+
+void Hachiko::Scripting::MainMenuManager::OnStart()
+{
+	_state = State::MAIN;
+	_state_changed = true;
+}
+
+void Hachiko::Scripting::MainMenuManager::OnUpdate()
+{
+	switch (_state)
+	{
+		case Hachiko::Scripting::MainMenuManager::State::MAIN:
+			OnUpdateMain();
+		break;
+		
+		case Hachiko::Scripting::MainMenuManager::State::SETTINGS:
+			OnUpdateSettings();
+		break;
+		
+		case Hachiko::Scripting::MainMenuManager::State::CREDITS:
+			OnUpdateCredits();
+		break;
+		
+		default:
+			HE_LOG("This should never happen.");
+		break;
+	}
+}
+
+void Hachiko::Scripting::MainMenuManager::OnUpdateMain()
+{
+	if (_state_changed)
+	{
+		// Common:
+		_main_background->SetActive(true);
+
+		// Main:
+		_button_background->SetActive(true);
+		_button_play->GetGameObject()->SetActive(true);
+		_button_settings->GetGameObject()->SetActive(true);
+		_button_quit->GetGameObject()->SetActive(true);
+		_button_credits->GetGameObject()->SetActive(true);
+
+		// Other:
+		_settings->SetActive(false);
+		_credits->SetActive(false);
+		_button_back->GetGameObject()->SetActive(false);
+
+		_state_changed = false;
+	}
+
+	if (_button_play->IsSelected())
+	{
+		SceneManagement::SwitchScene("Assets/Scenes/game.scene");
```

Comment:
Could uppercase cause any trouble?

Reply:
Maybe would cause trouble, right now, since we moved on to the new resource manager, we haven't recreated the old scenes with scripts on them. I'm changing it to lowercase with the existing scene names, in the following pull request after this, I will recreate old scenes we had.